controller/appcontroller.go

```diff
@@ -1959,8 +1959,11 @@ func alreadyAttemptedSync(app *appv1.Application, commitSHA string, commitSHAsMS
 			return false, ""
 		}
 	} else {
-		if app.Status.OperationState.SyncResult.Revision != commitSHA {
-			return false, ""
+		manifestsChangedMap := app.Status.Sync.ManifestsChanged
+		if manifestsChangedMap[commitSHA] {
+			if app.Status.OperationState.SyncResult.Revision != commitSHA {
```

Comment:
make fix for multi-sourced also

Reply:
I will do it in next PR, when will push to OSS. I am doing it for priceline right now

Reply:
i will do it when will open pr to oss, for now it is optional